scripts/update_specs_from_pricing.py

```diff
@@ -90,52 +87,27 @@ def get_paginator(service):
 
 
 def get_ec2_pricing():
-    results = {}
-    for page in get_paginator('AmazonEC2'):
-        for price_item in page.get('PriceList', []):
-            products = json.loads(price_item)
-            product = products.get('product', {})
-            if product:
-                if product.get('attributes').get('location') in region_exceptions:
-                    continue
-                if product.get('productFamily') in ['Compute Instance', 'Compute Instance (bare metal)']:
-                    if not results.get(region_map[product.get('attributes').get('location')]):
-                        results[region_map[product.get('attributes').get('location')]] = set()
-                    results[region_map[product.get('attributes').get('location')]].add(
-                        product.get('attributes').get('instanceType')
-                    )
-    return results
+    return get_results('AmazonEC2', ['Compute Instance', 'Compute Instance (bare metal)'])
 
 
 def get_redshift_pricing():
-    results = {}
-    for page in get_paginator('AmazonRedshift'):
-        for price_item in page.get('PriceList', []):
-            products = json.loads(price_item)
-            product = products.get('product', {})
-            if product:
-                if product.get('productFamily') == 'Compute Instance':
-                    if not results.get(region_map[product.get('attributes').get('location')]):
-                        results[region_map[product.get('attributes').get('location')]] = set()
-                    results[region_map[product.get('attributes').get('location')]].add(
-                        product.get('attributes').get('instanceType')
-                    )
-    return results
+    return get_results('AmazonRedshift', ['Compute Instance'])
 
 
 def get_dax_pricing():
+    service = 'AmazonDAX'
+    product_families = 'DAX'
     results = {}
-    for page in get_paginator('AmazonDAX'):
+    for page in get_paginator(service):
         for price_item in page.get('PriceList', []):
             products = json.loads(price_item)
             product = products.get('product', {})
             if product:
-                if product.get('productFamily') == 'DAX':
+                if product.get('productFamily') in product_families:
                     if not results.get(region_map[product.get('attributes').get('location')]):
                         results[region_map[product.get('attributes').get('location')]] = set()
-                    usage_type = product.get('attributes').get('usagetype').split(':')[1]
                     results[region_map[product.get('attributes').get('location')]].add(
-                        usage_type
+                        product.get('attributes').get('usagetype').split(':')[1]
```

Comment:
[`instanceType` missing the `dax.` prefix](https://pricing.us-east-1.amazonaws.com/offers/v1.0/aws/AmazonDAX/current/index.json)